### PR TITLE
Update Training Operators for 1.4

### LINKED ISSUE #1971: Sync Training WG manifests for KF 1.4-RC0
Sub issue from #1962 

We need to update the manifests provided by the Training WG with this repo, before cutting the first RC of manifests.

cc @kubeflow/wg-training-leads

### PR DESCRIPTION
**Pull request by kimwnasptd**
**Which issue is resolved by this Pull Request:**
Resolves #1971 

This PR:
1. Removes the previous `apps/*-job` directories
2. Creates a new `apps/training-operators/upstream` dir
3. Copies the manifests from https://github.com/kubeflow/tf-operator/tree/v1.3.0-alpha.2/manifests
4. Update the example kustomization to use the new folder structure

cc @kubeflow/wg-training-leads 
/cc @elikatsis @yanniszark

### CODE REVIEW

File: apps/mpi-job/upstream/base/deployment.yaml [hunk 1]
```diff
@@ -1,27 +0,0 @@
-apiVersion: apps/v1
```
Comment (andreyvelich) [inline apps/mpi-job/upstream/base/deployment.yaml:1]: I think MPIJob will not be part of Unify Training operator in Kubeflow 1.4, so we should keep these manifests.

cc @kubeflow/wg-training-leads

**Review by andreyvelich — COMMENTED** [review]
Thank you for taking this @kimwnasptd!

**Review by terrytangyuan — COMMENTED** [review]
Yes we should keep the existing MPI manfiests.

File: apps/mpi-job/upstream/base/deployment.yaml (hunk 1 above)
Comment (alculquicondor) [reply apps/mpi-job/upstream/base/deployment.yaml:1]: sg

**Comment by johnugeorge** [comment]
One minor comment
How about `training-operator` instead of  `training-operators`?

File: apps/mpi-job/upstream/base/deployment.yaml (hunk 1 above)
Comment (annajung) [reply apps/mpi-job/upstream/base/deployment.yaml:1]: @mkbhanda @DavidSpek @kimwnasptd @RFMVasconcelos 

It was brought up during the community meeting Aug 17th that MPI-Operator release should be kept separate for the 1.4 release

Comment (Jeffwan) [reply apps/mpi-job/upstream/base/deployment.yaml:1]: # MPI Operator
- ../apps/mpi-job/upstream/overlays/kubeflow

Let's keep this one

**Comment by Jeffwan** [comment]
Please help add `mpi-operator` back and let's ask @terrytangyuan to confirm manifests

**Review by terrytangyuan — APPROVED** [review]
LGTM thanks!

**Comment by johnugeorge** [comment]
LGTM
/cc @Jeffwan

**Comment by Jeffwan** [comment]
LGTM. Thanks

/lgtm

**Pull request merged**